src/parser.rs

```diff
@@ -335,24 +340,25 @@ impl<'tokens, 'src> Parser<'tokens, 'src> {
             {
               items.push(Item::Set(self.parse_set()?));
             } else {
-              items.push(Item::Recipe(self.parse_recipe(doc, false)?));
+              items.push(Item::Recipe(self.parse_recipe(&mut doc, false)?));
             },
           _ =>
             if self.next_are(&[Identifier, Equals]) {
               return Err(self.get(1)?.error(CompilationErrorKind::DeprecatedEquals));
             } else if self.next_are(&[Identifier, ColonEquals]) {
               items.push(Item::Assignment(self.parse_assignment(false)?));
             } else {
-              items.push(Item::Recipe(self.parse_recipe(doc, false)?));
+              items.push(Item::Recipe(self.parse_recipe(&mut doc, false)?));
             },
         }
       } else if self.accepted(At)? {
-        items.push(Item::Recipe(self.parse_recipe(doc, true)?));
+        items.push(Item::Recipe(self.parse_recipe(&mut doc, true)?));
       } else {
         return Err(self.unexpected_token()?);
       }
 
-      if next.kind != Comment {
+      if next.kind == Identifier && doc.is_some() {
+        items.insert(items.len() - 1, Item::Comment(doc.unwrap()));
         doc = None;
```

Comment:
I think we can avoid the unwrap with an `if let`:

```suggestion
      if next.kind == Identifier {
        if let Some(doc) = doc {
          items.insert(items.len() - 1, Item::Comment(doc));
          doc = None;
       }
```